### ISSUE TITLE
Use of mutex not enforced for data shared between threads

### ISSUE THREAD

**Issue by fisherro** [cppalliance/safe-cpp#65]
I was able to rewrite the example from section 1.5.3 of the proposal to eliminate the mutex and replace the arc with a box. This resulted in no error and allowed unsafe modification of shared data from multiple treads.

https://godbolt.org/z/aK167fGoM

```
#feature on safety
#include <https://raw.githubusercontent.com/cppalliance/safe-cpp/master/libsafecxx/single-header/std2.h?token=$(date%20+%s)>
#include <chrono>

using namespace std2;

void entry_point(box<string> data, int thread_id) safe {
  mut data->append("🔥");
  println(*data);
  unsafe { std::this_thread::sleep_for(std::chrono::milliseconds(10)); }
}

int main() safe {
  box<string> shared_data(string("Hello world - "));

  vector<thread> threads { };
  for(int i : 10)
    mut threads.push_back(thread(entry_point, cpy shared_data, i));
  
  for(thread t : rel threads)
    t rel.join();
}
```

**Comment by seanbaxter** [comment]
Thanks. It looks like someone @cmazakas didn't delete Box's copy constructor.

**Comment by seanbaxter** [comment]
Fixed in 9c39fc3c6c38c7. Adds deleted copy/borrow ctor to std2::box.
Thank you for reporting, @fisherro .
